### ISSUE TITLE
Multiple documents link to outdated static links

### ISSUE THREAD

**Issue by rwlee** [NVIDIA/spark-rapids-examples#218]
**Describe the bug**
Documents, examples, scripts, and jar repository links are largely out of sync -- leads to inconsistent examples and hard to follow instructions when trying to run with the latest functionality.

`grep -E -r "22\.02|22\.04|22\.06" .` ran at the base of a clean `branch-22.08` repository

```
./examples/UDF-Examples/RAPIDS-accelerated-UDFs/pom.xml:        <rapids4spark.version>22.06.0</rapids4spark.version>
./examples/UDF-Examples/RAPIDS-accelerated-UDFs/README.md:[rapids-4-spark_2.12-22.06.0.jar](https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark_2.12/22.06.0/rapids-4-spark_2.12-22.06.0.jar)
./examples/UDF-Examples/Spark-cuSpatial/notebooks/cuspatial_sample_standalone.ipynb:    "jarsPath = os.getenv(\"JARS_PATH\", \"/data/cuspatial_data/jars/rapids-4-spark_2.12-22.06.0.jar,/data/cuspatial_data/jars/spark-cuspatial-22.08.0-SNAPSHOT.jar\")\n",
./examples/UDF-Examples/Spark-cuSpatial/pom.xml:    <rapids.version>22.06.0</rapids.version>
./examples/UDF-Examples/Spark-cuSpatial/gpu-run.sh:JARS_PATH=${JARS_PATH:-$JARS/rapids-4-spark_2.12-22.06.0.jar,$JARS/spark-cuspatial-22.08.0-SNAPSHOT.jar}
./examples/UDF-Examples/Spark-cuSpatial/README.md:    conda install -c rapidsai -c nvidia -c conda-forge  -c defaults libcuspatial=22.06
./examples/UDF-Examples/Spark-cuSpatial/README.md:   * [Spark RAPIDS JAR v22.06.0](https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark_2.12/22.06.0/rapids-4-spark_2.12-22.06.0.jar) or above
./examples/UDF-Examples/Spark-cuSpatial/src/main/native/CMakeLists.txt:project(SPATIALUDJNI VERSION 22.06.0 LANGUAGES C CXX CUDA)
./examples/UDF-Examples/Spark-cuSpatial/Dockerfile.awsdb:RUN conda install -c rapidsai-nightly -c nvidia -c conda-forge  -c defaults libcuspatial=22.06
./examples/UDF-Examples/Spark-cuSpatial/Dockerfile:RUN conda install -c rapidsai -c nvidia -c conda-forge -c defaults libcuspatial=22.06 python=3.8 -y
./examples/ML+DL-Examples/Spark-cuML/pca/README.md:[rapids-4-spark-ml_2.12-22.02.0-cuda11.jar](https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark-ml_2.12/22.02.0/rapids-4-spark-ml_2.12-22.02.0-cuda11.jar)
./examples/ML+DL-Examples/Spark-cuML/pca/README.md:[rapids-4-spark_2.12-22.06.0.jar](https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark_2.12/22.06.0/rapids-4-spark_2.12-22.06.0.jar)
./examples/ML+DL-Examples/Spark-cuML/pca/README.md:    RAPIDS_ML_JAR=PATH_TO_rapids-4-spark-ml_2.12-22.02.0-cuda11.jar
./examples/ML+DL-Examples/Spark-cuML/pca/README.md:    PLUGIN_JAR=PATH_TO_rapids-4-spark_2.12-22.06.0.jar
./examples/ML+DL-Examples/Spark-cuML/pca/Dockerfile:ARG BRANCH_VER=22.06
./examples/SQL+DF-Examples/micro-benchmarks/notebooks/micro-benchmarks-gpu.ipynb:    "RAPIDS_JAR = os.getenv(\"RAPIDS_JAR\", \"/your-path/rapids-4-spark_2.12-22.06.0.jar\")\n"
./examples/XGBoost-Examples/README.md:For PySpark based XGBoost, please refer to the [Spark-RAPIDS-examples 22.04 branch](https://github.com/NVIDIA/spark-rapids-examples/tree/branch-22.04) that
./examples/XGBoost-Examples/agaricus/notebooks/python/agaricus-gpu.ipynb:    "Note: For PySpark based XGBoost, please refer to the [Spark-RAPIDS-examples 22.04 branch](https://github.com/NVIDIA/spark-rapids-examples/tree/branch-22.04) that\n",
./examples/XGBoost-Examples/taxi/notebooks/python/cv-taxi-gpu.ipynb:    "Note: For PySpark based XGBoost, please refer to the [Spark-RAPIDS-examples 22.04 branch](https://github.com/NVIDIA/spark-rapids-examples/tree/branch-22.04) that\n",
./examples/XGBoost-Examples/taxi/notebooks/python/taxi-gpu.ipynb:    "Note: For PySpark based XGBoost, please refer to the [Spark-RAPIDS-examples 22.04 branch](https://github.com/NVIDIA/spark-rapids-examples/tree/branch-22.04) that\n",
./examples/XGBoost-Examples/taxi/notebooks/python/taxi-ETL.ipynb:    "* [rapids-4-spark_2.12-22.06.0.jar](https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark_2.12/22.06.0/rapids-4-spark_2.12-22.06.0.jar)\n",
./examples/XGBoost-Examples/taxi/notebooks/python/taxi-ETL.ipynb:    "$ export SPARK_JARS=rapids-4-spark_2.12-22.06.0.jar\n",
./examples/XGBoost-Examples/taxi/notebooks/scala/taxi-ETL.ipynb:    "* [rapids-4-spark_2.12-22.06.0.jar](https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark_2.12/22.06.0/rapids-4-spark_2.12-22.06.0.jar)\n",
./examples/XGBoost-Examples/taxi/notebooks/scala/taxi-ETL.ipynb:    "$ export SPARK_JARS=rapids-4-spark_2.12-22.06.0.jar\n",
./examples/XGBoost-Examples/mortgage/notebooks/python/MortgageETL+XGBoost.ipynb:    "Note: For PySpark based XGBoost, please refer to the [Spark-RAPIDS-examples 22.04 branch](https://github.com/NVIDIA/spark-rapids-examples/tree/branch-22.04) that\n",
./examples/XGBoost-Examples/mortgage/notebooks/python/mortgage-gpu.ipynb:    "Note: For PySpark based XGBoost, please refer to the [Spark-RAPIDS-examples 22.04 branch](https://github.com/NVIDIA/spark-rapids-examples/tree/branch-22.04) that\n",
./examples/XGBoost-Examples/mortgage/notebooks/python/MortgageETL.ipynb:    "* [rapids-4-spark_2.12-22.06.0.jar](https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark_2.12/22.06.0/rapids-4-spark_2.12-22.06.0.jar)\n",
./examples/XGBoost-Examples/mortgage/notebooks/python/MortgageETL.ipynb:    "$ export SPARK_JARS=rapids-4-spark_2.12-22.06.0.jar\n",
./examples/XGBoost-Examples/mortgage/notebooks/python/cv-mortgage-gpu.ipynb:    "Note: For PySpark based XGBoost, please refer to the [Spark-RAPIDS-examples 22.04 branch](https://github.com/NVIDIA/spark-rapids-examples/tree/branch-22.04) that\n",
./examples/XGBoost-Examples/mortgage/notebooks/scala/mortgage-ETL.ipynb:    "* [rapids-4-spark_2.12-22.06.0.jar](https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark_2.12/22.06.0/rapids-4-spark_2.12-22.06.0.jar)\n",
./examples/XGBoost-Examples/mortgage/notebooks/scala/mortgage-ETL.ipynb:    "$ export SPARK_JARS=rapids-4-spark_2.12-22.06.0.jar\n",
./docs/get-started/xgboost-examples/csp/databricks/generate-init-script.ipynb:    "sudo wget -O rapids-4-spark_2.12-22.06.0.jar https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark_2.12/22.06.0/rapids-4-spark_2.12-22.06.0.jar\n",
./docs/get-started/xgboost-examples/csp/databricks/generate-init-script.ipynb:    "sudo cp /dbfs/FileStore/jars/rapids-4-spark_2.12-22.06.0.jar /databricks/jars/\n",
./docs/get-started/xgboost-examples/csp/databricks/generate-init-script.ipynb:    "4. Import the mortgage example notebook from `https://github.com/NVIDIA/spark-rapids-examples/blob/branch-22.06/examples/XGBoost-Examples/mortgage/notebooks/python/mortgage-gpu.ipynb`\n",
./docs/get-started/xgboost-examples/csp/databricks/generate-init-script-10.4.ipynb:    "sudo wget -O rapids-4-spark_2.12-22.06.0.jar https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark_2.12/22.06.0/rapids-4-spark_2.12-22.06.0.jar\n",
./docs/get-started/xgboost-examples/csp/databricks/generate-init-script-10.4.ipynb:    "sudo cp /dbfs/FileStore/jars/rapids-4-spark_2.12-22.06.0.jar /databricks/jars/\n",
./docs/get-started/xgboost-examples/csp/databricks/generate-init-script-10.4.ipynb:    "4. Import the mortgage example notebook from `https://github.com/NVIDIA/spark-rapids-examples/blob/branch-22.06/examples/XGBoost-Examples/mortgage/notebooks/python/mortgage-gpu.ipynb`\n",
./docs/get-started/xgboost-examples/csp/databricks/databricks.md:    installed.  Users will need to use 22.06.0 or later on Databricks 10.4 LTS ML. In this case use
./docs/get-started/xgboost-examples/prepare-package-data/preparation-scala.md:   * [RAPIDS Spark Package](https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark_2.12/22.06.0/rapids-4-spark_2.12-22.06.0.jar)
./docs/get-started/xgboost-examples/prepare-package-data/preparation-scala.md:export RAPIDS_JAR=${SPARK_XGBOOST_DIR}/rapids-4-spark_2.12-22.06.0.jar
./docs/get-started/xgboost-examples/prepare-package-data/preparation-python.md:   * [RAPIDS Spark Package](https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark_2.12/22.06.0/rapids-4-spark_2.12-22.06.0.jar)
./docs/get-started/xgboost-examples/prepare-package-data/preparation-python.md:export RAPIDS_JAR=${SPARK_XGBOOST_DIR}/rapids-4-spark_2.12-22.06.0.jar
./docs/get-started/xgboost-examples/on-prem-cluster/kubernetes-scala.md:wget https://github.com/NVIDIA/spark-rapids-examples/raw/branch-22.06/dockerfile/Dockerfile
./docs/get-started/xgboost-examples/notebook/python-notebook.md:[Spark-RAPIDS-examples 22.04 branch](https://github.com/NVIDIA/spark-rapids-examples/blob/branch-22.04/docs/get-started/xgboost-examples/notebook/python-notebook.md) 
```

**Comment by rwlee** [comment]
Some of these issues are deeper issues requiring more complete resolutions.
For example, `spark-rapids-examples/examples/ML+DL-Examples/Spark-cuML/pca/README.md` contains a link to a maven repository that does not have a version `22.04` or above.

**Comment by nvliyuan** [comment]
yes, there are multi dependencies in this repo and it may be confusing.

- Since the last [PCA release](https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark-ml_2.12/) is v22.02, so we don't update in the following release;
- and since dmlc1.6.1 don't support XGBoost python api, so if customers want to test with python code, we refer to v22.04 for the old codes;
- all v22.06 should be updated to v22.08 during the release, we just merged this [pr](https://github.com/NVIDIA/spark-rapids-examples/pull/215) and merged branch-22.08 to main, so in branch-22.08 and main branch there should be the right version; maybe we can update the version at the time when we create the next-release-dev branch to make it more clear.

**Comment by nvliyuan** [comment]
close the issue for now since reconstructing the example repo, will reopen if still needed
